### DISCUSSION TITLE
Have you learned of a way to mark solved the cube state by writing via a bluetooth characteristic?

### DISCUSSION THREAD

**Discussion by FoxMcWeezer** [cs0x7f/cstimer#348] [Q&A] [answered]
I am looking at markSolved in bluetoothutil.js but it looks like this only updates/recalibrates the csTimer's representation of the cube and stored to localStorage. We know it is possible to write to the cube's state since this is what CubeStation does when a user resets to solved state. The cube can be connected back to 3rd party services like csTimer, and the solved state is read, proving CubeStation indeeds writes to the internals of the cube. What do you think about this @cs0x7f

**Comment by cs0x7f** [comment, marked as answer]
csTimer is not only for a specific brand of smart cube, so it needs to be designed to minimize the dependence on specific interfaces. For the markSolved function, I think the simplest implementation is sufficient, even if there is an interface that can modify the state of the smart cube, additional interface calls are not necessary, and will reduce the overall robustness.

**Reply by FoxMcWeezer** [reply]
My mistake. I thought it wasn't in the code but I didn't look hard enough. I see you have implemented the ability to mark the cube solved. It works great. Thanks

```
return this.v2sendRequest([10, 5, 57, 119, 0, 0, 1, 35, 69, 103, 137, 171, 0, 0, 0, 0, 0, 0, 0, 0]);
```
